dao/authentication_dao_test.go

```diff
@@ -320,7 +321,6 @@ func TestAuthFromVault(t *testing.T) {
 		Name:                    "test-vault-auth",
 		AuthType:                "test-authtype",
 		Username:                "my-username",
-		Password:                "my-password",
```

Comment:
I would use `util.StringRef("my-password")` here...

Reply:
The main reason I didn't do that is because due to the way this unit test works the `toVaultMap` function would return a map[string]interface{} that then has `data["password"] = *"my-password"`, which isn't the way that the actual vault would be returning.

Instead a "real" vault would return a string OR null, so we're checking for `nil` before casting it to a `string`.